geowebcache/wms/src/main/java/org/geowebcache/service/wms/WMSService.java

```diff
@@ -302,7 +302,14 @@ public void handleRequest(Conveyor conv) throws GeoWebCacheException {
         ConveyorTile tile = (ConveyorTile) conv;
 
         String servletPrefix = null;
-        if (controller != null) servletPrefix = controller.getServletPrefix();
+
+        if (controller != null) {
+            servletPrefix = controller.getServletPrefix();
+            XMLConfiguration mainConfiguration2 =
+                    (XMLConfiguration) controller.getMainConfiguration();
+            if (mainConfiguration2 != null)
+                tld.setServiceInformation(mainConfiguration2.getServiceInformation());
```

Comment:
Why is the default service information being set in response to a tile request?

Reply:
I'd be happy for Spring to inject the information into the TLD but I couldn't see any way to make that work, is there an example of that in another class I can look at? 

I'm happy to look at adding a test

Reply:
Well, the TLD is already doing that via two different mechanisms It gets the GridSetBroker as a constructor parameter.  geowebcache-core-context.xml defines the TLD bean so you would then have to update that to tell Spring it needs a reference to the XMLConfiguration bean.

The other method is that the TLD is defined as being ApplicationContextAware which means SPring gives it a refernece to the context, which TLD then uses to extract whatever beans it wants, in this case it looks up all the intances of LayerConfiguration.  Look in the afterPropertiesSet method.

The first method is better practice and is nicer for unit testing but does involve an API change which requires a downstream change in GeoServer (updating the context XML file)  The second method avoids the API change but makes unit tests a bit clunkier as you have to mock the application context.  The second method also requires either coding the id of the BaseConfiguration bean into the TLD, or accounting for the possibility of there being more than one (With the constructor and context files you are telling spring explicitly which bean to use so you don't need to worry about that in the code)

Reply:
OK, I think I've cracked it - I went with the 2nd method so no downstream changes (I hope) but lots of fun with mocking :-)